iam-login-service/src/main/java/it/infn/mw/iam/api/account_linking/DefaultAccountLinkingService.java

```diff
@@ -151,7 +151,7 @@ public void linkX509Certificate(Principal authenticatedUser,
 
     Optional<IamX509Certificate> linkedCert = userAccount.getX509Certificates()
       .stream()
-      .filter(c -> c.getSubjectDn().equals(x509Credential.getSubject()))
+      .filter(c -> c.getIssuerDn().equals(x509Credential.getIssuer()))
```

Comment:
We should keep both.

Reply:
It's probably useless to set again subject and issuer at lines 160-161. And since these certificates are kept in a Set, it's probably also safer not to touch the fields that define the order in the Set.